src/test/java/com/epam/ShoppingCart.java

```diff
@@ -0,0 +1,33 @@
+package com.epam;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class ShoppingCart {
+
+    @Test
+    @DisplayName("Verify shopping cart is empty & summary steps are displayed")
+
+    public void dressesCount() {
+        System.setProperty("webdriver.chrome.driver", "chromedriver.exe");
+        WebDriver driver = new ChromeDriver();
+        driver.get("http://automationpractice.com/index.php");
+
+        driver.findElement(By.xpath("//div/a[@title='View my shopping cart']")).click();
+        Assertions.assertEquals("Your shopping cart is empty.", driver.findElement(By.xpath("//div/p[contains (@class, 'alert-warning')]")).getText());
+
+        driver.findElement(By.xpath("//ul/li[contains (@class, 'first')]"));
+        driver.findElement(By.xpath("//ul/li[contains (@class, 'second')]"));
+        driver.findElement(By.xpath("//ul/li[contains (@class, 'third')]"));
+        driver.findElement(By.xpath("//ul/li[contains (@class, 'four')]"));
+        driver.findElement(By.xpath("//ul/li[contains (@class, 'last')]"));
```

Comment:
this is a typically ok solution. You only have one assertion point for 5 checks, that seems quite ok.

there's a catch though. if the first fails, neither of the remaining ones will be asserted.
on the other hand, if the first fails, perhaps it's best not to assert to avoid false negative cases(maybe the first check is a prerequisite)

this is a constant dilemma how to verify multi-assertions.

but, what's important to know is our toolset. and in our toolset we have soft assertions. with that we can verify all without failing the test after the first. and once all of them are verified we can make a decision whether or not to fail the test.

the simplest solution is to collect the boolean values of all these check and then use the AND operator on them

but there are more sophisticated libraries out there [e.g. with TestNG](https://www.seleniumeasy.com/testng-tutorials/soft-asserts-in-testng-example)

Reply:
I have tried it with .isDisplayed() method, looks ok but maybe worth to discuss on our meetings